### DISCUSSION TITLE
Build libcurl with QUICHE and BoringSSL using cmake

### DISCUSSION THREAD

**Discussion by pkropachev** [curl/curl#12396] [Q&A] [answered]
Hello!

I faced with the following problem during attempt to build libcurl with QUICHE and BoringSSL using cmake:

```
cmake -DUSE_QUICHE=ON -DQUICHE_LIBRARY=$PWD/../../quiche/target/release -DQUICHE_INCLUDE_DIR=$PWD/../../quiche/quiche/include ..

...

-- Found QUICHE: /home/pk/projects/quiche/target/release  
CMake Error at CMakeLists.txt:712 (CheckQuicSupportInOpenSSL):
  Unknown CMake command "CheckQuicSupportInOpenSSL".
```

This is first problem. I can't find where `CheckQuicSupportInOpenSSL` is implemented in CMakeLists.txt.

I commented `CheckQuicSupportInOpenSSL` in CMakeLists.txt. After that I got the following output:

```
CMake Warning at /home/pk/projects/curl/build/CMakeFiles/CMakeScratch/TryCompile-jvB7Nv/CMakeLists.txt:22 (target_link_libraries):
  Target "cmTC_93563" requests linking to directory
  "/home/pk/projects/quiche/target/release".
  Targets may link only to libraries.  CMake is dropping the item.
```

**Comment by bagder** [comment]
If you insist on using cmake for this, you probably need to roll up your sleeves and be prepared to fix some breakage.

The documented way to build HTTP/3 support for curl uses configure and that is what we do and verify in CI jobs. The cmake setup for this I believe is intended to work but is not used nor verified on a regular basis so might have degraded.

**Reply by pkropachev** [reply, marked as answer]
Thanks!

It looks like the problem was already fixed in #12160 and #12162. I'll check on my side.
